.azure_pipelines/performance_check/run_performance_check.py

```diff
@@ -0,0 +1,282 @@
+
+
+def export_optimum_dynamic_quantization(onnx_model_path, model_root_path):
+    quantizer = ORTQuantizer.from_pretrained(onnx_model_path)
+    quantization_model_path = model_root_path / "optimum_dynamic_quantization"
+    dqconfig = AutoQuantizationConfig.avx512_vnni(is_static=False, per_channel=False)
+    quantizer.quantize(
+        save_dir=quantization_model_path,
+        quantization_config=dqconfig,
+    )
+
+
+def run_with_config(tool, olive_config, metric_res):
+    outputs = olive_run(olive_config)
+    if tool == "olive":
+        metric = str(list(list(outputs.values())[0].nodes.values())[0].metrics.value)
+    else:
+        metric = str(list(outputs.values())[0])
+    metric_dict = ast.literal_eval(metric)
+
+    for metric_name, metric_value in metric_dict.items():
+        if metric_name not in metric_res[tool]:
+            metric_res[tool][metric_name] = []
+        metric_res[tool][metric_name].append(metric_value)
+
+
+def run_perf_comparison(cur_dir, model_name, device, model_root_path, test_num):
+    print(f"Start running perf comparison on {model_name} model {test_num} times...")
+    model_list = ["hf_pytorch", "pytorch_compile", "onnx", "optimum_o1", "optimum_o2", "optimum_o3"]
+    if device == "gpu":
+        model_list.append("optimum_o4")
+    if device == "cpu":
+        model_list.append("optimum_dynamic_quantization")
+    metric_res = {}
+    config_json_path = cur_dir / "configs" / "perf.json"
+    for optimized_model in model_list:
+        metric_res[f"{optimized_model}"] = {}
+    metric_res["olive"] = {}
+    for i in range(test_num):
+        print(f"Start running {i} time...")
+        for optimized_model in model_list:
+            print(f"Start evaluating {optimized_model} model")
+            with open(config_json_path, "r") as fin:
+                olive_config = json.load(fin)
+                user_script_path = str(cur_dir / "user_scripts" / f"{model_name}.py")
+                if optimized_model == "onnx":
+                    olive_config["input_model"]["config"]["model_path"] = str(
+                        Path(model_root_path / optimized_model / "model.onnx")
+                    )
+                elif optimized_model == "optimum_dynamic_quantization":
+                    olive_config["input_model"]["config"]["model_path"] = str(
+                        Path(model_root_path / optimized_model / "model_quantized.onnx")
+                    )
+                elif optimized_model in ["optimum_o1", "optimum_o2", "optimum_o3", "optimum_o4"]:
+                    olive_config["input_model"]["config"]["model_path"] = str(
+                        Path(model_root_path / optimized_model / "model_optimized.onnx")
+                    )
+                elif optimized_model == "hf_pytorch":
+                    olive_config["input_model"]["type"] = "PyTorchModel"
+                    hf_config = {"hf_config": MODEL_NAME_TO_CONFIG_MAP[model_name]}
```

Comment:
I am interested in how the code works as expected since the hf_config is only being added for this case. For the other cases, there is no hf_config but the metrics are still working fine (when they don't have data_config to begin with). 

Maybe I am missing something here

Reply:
oh, looks like it might be a lucky instance where the `ACC_METRIC` and `LAT_METRIC` objects are modified by the first iteration. 
since the first iteration is `hf_pytorch`, the data config gets auto inserted from the input model and the other iterations use this. 

To be safe and correct, I think it might be better to update the metrics beforehand with the data_configs with something like:
```
ACC_METRIC["data_config"] = huggingface_config_template(model_name, task, **dataset)
```
That way the metrics already have the data config and don't rely on it being inserted by the first iteration.

Reply:
Good catch! I also noticed the metrics were different since this change, but I'm not sure if this is related how we feed data to metric (theoretically no but this is where the difference starts). I'll update this part logic later.

Reply:
During the workflow run validation we auto insert the data config from the input model if the input model has hf_config.dataset and the metric's `data_config` is `None`. 

Because the metrics added by reference, they are getting updated globally during the first run.

Reply:
It may cause longer Olive workflow run time but this shouldn't affect latency metric. Not sure why latency changes.

Reply:
I don't think the runtime gets affected since both accuracy and latency require data configs if there is no dataloader func. Also, both metrics will have the data_config auto-inserted. 

latency metric can try to get io_config from the onnx model itself but since they have dynamic shape, this would not work.